src/huggingface_hub/inference_api.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from typing import Dict, List, Optional, Union
+
+import requests
+
+from .hf_api import HfApi, HfFolder
+
+
+logger = logging.getLogger(__name__)
+
+
+ENDPOINT = "https://api-inference.huggingface.co/pipeline"
+
+ALL_TASKS = [
+    # NLP
+    "text-classification",
+    "token-classification",
+    "table-question-answering",
+    "question-answering",
+    "zero-shot-classification",
+    "translation",
+    "summarization",
+    "conversational",
+    "feature-extraction",
+    "text-generation",
+    "text2text-generation",
+    "fill-mask",
+    "sentence-similarity",
+    # Audio
+    "text-to-speech",
+    "automatic-speech-recognition",
+    "audio-source-separation",
+    "voice-activity-detection",
+    # Computer vision
+    "image-classification",
+    "object-detection",
+    "image-segmentation",
+]
+
+
+class InferenceApi:
+    """Client to configure requests and make calls to the HuggingFace Inference API.
+
+    Example:
+
+            >>> from huggingface_hub.inference_api import InferenceApi
+
+            >>> # Mask-fill example
+            >>> api = InferenceApi("bert-base-uncased")
+            >>> api(inputs="The goal of life is [MASK].")
+            >>> >> [{'sequence': 'the goal of life is life.', 'score': 0.10933292657136917, 'token': 2166, 'token_str': 'life'}]
+
+
+            >>> # Question Answering example
+            >>> api = InferenceApi("deepset/roberta-base-squad2")
+            >>> inputs = {"question":"What's my name?", "context":"My name is Clara and I live in Berkeley."}
+            >>> api(inputs)
+            >>> >> {'score': 0.9326569437980652, 'start': 11, 'end': 16, 'answer': 'Clara'}
+
+            >>> # Zero-shot example
+            >>> api = InferenceApi("typeform/distilbert-base-uncased-mnli")
+            >>> inputs = "Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!"
+            >>> params = {"candidate_labels":["refund", "legal", "faq"]}
+            >>> api(inputs, params)
+            >>> >> {'sequence': 'Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!', 'labels': ['refund', 'faq', 'legal'], 'scores': [0.9378499388694763, 0.04914155602455139, 0.013008488342165947]}
+    """
+
+    def __init__(
+        self,
+        repoId: str,
```

Comment:
```suggestion
        repo_id: str,
```

`repoId` looks weird in Python and isn't consistent with rest of the codebase